crates/orchestrator/src/queue/mod.rs

```diff
@@ -9,77 +9,119 @@ use color_eyre::Result as EyreResult;
 use lazy_static::lazy_static;
 use mockall::automock;
 use omniqueue::{Delivery, QueueError};
+use strum_macros::Display;
 
 use crate::config::Config;
 use crate::jobs::JobError;
 use crate::setup::SetupConfig;
 
+#[derive(Display, Debug, Clone, PartialEq, Eq)]
+pub enum QueueType {
+    #[strum(serialize = "snos_job_processing")]
+    SnosJobProcessing,
+    #[strum(serialize = "snos_job_verification")]
+    SnosJobVerification,
+    #[strum(serialize = "proving_job_processing")]
+    ProvingJobProcessing,
+    #[strum(serialize = "proving_job_verification")]
+    ProvingJobVerification,
+    #[strum(serialize = "proof_registration_job_processing")]
+    ProofRegistrationJobProcessing,
+    #[strum(serialize = "proof_registration_job_verification")]
+    ProofRegistrationJobVerification,
+    #[strum(serialize = "data_submission_job_processing")]
+    DataSubmissionJobProcessing,
+    #[strum(serialize = "data_submission_job_verification")]
+    DataSubmissionJobVerification,
+    #[strum(serialize = "update_state_job_processing")]
+    UpdateStateJobProcessing,
+    #[strum(serialize = "update_state_job_verification")]
+    UpdateStateJobVerification,
+    #[strum(serialize = "job_handle_failure")]
+    JobHandleFailure,
+    #[strum(serialize = "worker_trigger")]
+    WorkerTrigger,
+}
+
+impl QueueType {
```

Comment:
does strum macros have something that can help with this

Reply:
Valid, using EnumIter